### DISCUSSION TITLE
Unable to edit firstName, lastName, and email fields in latest Keycloak version

### DISCUSSION THREAD

**Discussion by abcgco** [keycloak/keycloak#30822] [Q&A]
### Description
After updating to the latest version of Keycloak, we have encountered an issue where we can no longer edit the `firstName`, `lastName`, and `email` fields for users through the admin console or via the Admin REST API. These fields appear to be set as read-only, and we are unsure how to modify this setting.

### Steps to Reproduce
1. Log in to the Keycloak admin console.
2. Navigate to Users and select any user.
3. Try to edit the `firstName`, `lastName`, or `email` fields.
4. Fields are displayed as read-only and cannot be modified.

### Expected Behavior
We expect to be able to edit these fields as our business process involves regular updates to user profiles.

### Actual Behavior
The fields `firstName`, `lastName`, and `email` are read-only and cannot be edited either through the admin console or via API calls.

### Keycloak Version
Indicate the version of Keycloak you are using (e.g., 15.0.2).

### Additional Information
We have checked the following settings:
- User profile configuration under Realm Settings does not show these fields as read-only.
- No custom scripts or policies are applied that might enforce these fields as read-only.

We suspect that there may be a new configuration setting or policy introduced in the latest updates that we are not aware of. Any guidance or documentation on how to adjust these settings would be greatly appreciated.

{"errors":[{"field":"lastName","errorMessage":"error-user-attribute-read-only","params":["lastName"]},{"field":"firstName","errorMessage":"error-user-attribute-read-only","params":["firstName"]},{"field":"email","errorMessage":"error-user-attribute-read-only","params":["email"]}]}

### Questions
1. Has there been a change in the latest versions that affects the editability of these fields?
2. What steps can we take to make these fields editable again?

Thank you for your assistance!

**Comment by pedroigor** [comment]
> Has there been a change in the latest versions that affects the editability of these fields?

AFAIK, no change that would block you from writing those attributes when they are not marked as read-only in the configuration.

> What steps can we take to make these fields editable again?

I'm unsure. I can't reproduce the issue and we have tests that are constantly updating those fields. Perhaps you can create an issue and provide more details about your deployment and the steps to reproduce.

**Comment by saman-jafari** [comment]
I encountered a similar issue and found a workaround that might help. If you're using any Federation or User Storage providers, try disabling and then re-enabling them. It seems that sometimes the system gets stuck, particularly when **read-only** is enabled by Federations, and this state is shared across all users. I'm not entirely sure why this happens, but this method worked for us.

In our case, the problem was with our user federation configured with Active Directory (AD). We had it set to read-only, and simply disabling and re-enabling the federation resolved the issue. Interestingly, we couldn't consistently reproduce the problem; out of 12 instances, 3 were affected. Our tests covered upgrades from versions 23 to 25, 24 to 25, and even 25.0.1 to 25.0.2. The issue appears to be flaky—sometimes it occurs, and sometimes it doesn't.

So, I suggest trying the classic "turn it off and on again" approach :grin: by disabling and then re-enabling your federations. Please let me know if this resolves your problem.

**Comment by sebastianBoehringer** [comment]
This sounds like [30476](https://github.com/keycloak/keycloak/issues/30476). We had the issue described there. Simply disabling and enabling our user federation did not solve the issue permanently. But changing their edit mode from "READ_ONLY" to "UNSYNCED" did the trick. (This was the workaround described in the issue)
